src/containers/Board/DetailBoard/DetailBoard.tsx

```diff
@@ -0,0 +1,106 @@
+import { useRouter } from "next/router";
+import { useMutation } from "@tanstack/react-query";
+import { useEffect, useState } from "react";
+import { AxiosError } from "axios";
+
+import * as S from "./emotion";
+import { boardsAPI } from "@/apis";
+
+export default function DetailBoard() {
+  const router = useRouter();
+  const { postId } = router.query;
+
+  const [post, setPost] = useState<Swagger.FreePostDetailResponseDto | null>(
+    null
+  );
+
+  useEffect(() => {
+    mutate();
+  }, []);
+
+  const checkApi = async (type: "get" | "remove") => {
+    let response;
+
+    switch (router.query.type) {
+      case "free":
+        if (type === "get") {
+          response = await boardsAPI.freePost.getPost({
+            freePostId: Number(postId),
+          });
+        } else if (type === "remove") {
+          response = await boardsAPI.freePost.removePost({
+            freePostId: Number(postId),
+          });
+        }
+        break;
+
+      case "notice":
+        if (type === "get") {
+          response = await boardsAPI.noticeBoard.getPost({
+            noticePostId: Number(postId),
+          });
+        } else if (type === "remove") {
+          response = await boardsAPI.noticeBoard.removePost({
+            noticePostId: Number(postId),
+          });
+        }
+        break;
+    }
+    return response;
+  };
+
+  const { mutate } = useMutation({
+    mutationFn: async () => {
+      // const response = await checkApi('get', router.query.type as string);
+      const response = await boardsAPI.freePost.getPost({
+        freePostId: Number(postId),
+      });
+
+      // const response = await boardsAPI.noticePost.getPost({
+      //   noticePostId: Number(postId),
+      // });
+
+      return response;
+    },
+    mutationKey: ["post", post],
+    onSuccess(data) {
+      setPost(data);
+    },
+    onError(error) {
+      const err = error as AxiosError<SwaggerError.GeneralApiError>;
+      alert(err.response?.data.message);
+    },
+  });
```

Comment:
```suggestion
  const { data } = useQuery({
    queryFn: async () => {
      // const response = await checkApi('get', router.query.type as string);
      const response = await boardsAPI.freePost.getPost({
        freePostId: Number(postId),
      });

      // const response = await boardsAPI.noticePost.getPost({
      //   noticePostId: Number(postId),
      // });

      return response;
    },
    queryKey: ["post", postId],
    enabled: postId !== undefined
  });
```

이건 페이지 입장했을때 데이터를 바로 가져와야 하기 때문에 useMutaion보다는 useQuery가 더 적절해보이네요